ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/DebugDataDumperTest.java

```diff
@@ -0,0 +1,151 @@
+    checkFileNotExist(expectedFile);
+  }
+
+  @Test
+  void saveGossipRejectedMessageToFile_shouldSaveToFile(@TempDir Path tempDir) {
+    final DebugDataDumper manager = new DebugDataDumper(tempDir, true);
+    final Bytes messageBytes = dataStructureUtil.stateBuilderPhase0().build().sszSerialize();
+    final String arrivalTimestamp = timeProvider.getTimeInMillis().toString();
+    final String topic = "test_topic";
+    manager.saveGossipRejectedMessageToFile("test_topic", arrivalTimestamp, messageBytes);
+
+    final String fileName = String.format("%s_%s.ssz", arrivalTimestamp, topic);
+    final Path expectedFile =
+        tempDir.resolve("gossip_messages").resolve("rejected").resolve(topic).resolve(fileName);
+    checkBytesSavedToFile(expectedFile, messageBytes);
+  }
+
+  @Test
+  void saveGossipRejectedMessageToFile_shouldNotSaveToFileWhenDisabled(@TempDir Path tempDir) {
+    final DebugDataDumper manager = new DebugDataDumper(tempDir, false);
+    final Bytes messageBytes = dataStructureUtil.stateBuilderPhase0().build().sszSerialize();
+    final String arrivalTimestamp = timeProvider.getTimeInMillis().toString();
+    manager.saveGossipRejectedMessageToFile("test_topic", arrivalTimestamp, messageBytes);
+    assertThat(manager.isEnabled()).isFalse();
+
+    final String fileName = String.format("%s_%s.ssz", arrivalTimestamp, "test_topic");
+    final Path expectedFile =
+        tempDir.resolve("gossip_messages").resolve("rejected").resolve(fileName);
+    checkFileNotExist(expectedFile);
+  }
+
+  @Test
+  void saveInvalidBlockToFile_shouldSaveToFile(@TempDir Path tempDir) {
+    final DebugDataDumper manager = new DebugDataDumper(tempDir, true);
+    final SignedBeaconBlock block = dataStructureUtil.randomSignedBeaconBlock();
+    manager.saveInvalidBlockToFile(block.getSlot(), block.getRoot(), block.sszSerialize());
+
+    final String fileName =
+        String.format(
+            "slot%s_root%s.ssz", block.getSlot(), block.getRoot().toUnprefixedHexString());
+    final Path expectedFile = tempDir.resolve("invalid_blocks").resolve(fileName);
+    checkBytesSavedToFile(expectedFile, block.sszSerialize());
+  }
+
+  @Test
+  void saveInvalidBlockToFile_shouldNotSaveToFileWhenDisabled(@TempDir Path tempDir) {
+    final DebugDataDumper manager = new DebugDataDumper(tempDir, false);
+    final SignedBeaconBlock block = dataStructureUtil.randomSignedBeaconBlock();
+    manager.saveInvalidBlockToFile(block.getSlot(), block.getRoot(), block.sszSerialize());
+    assertThat(manager.isEnabled()).isFalse();
+
+    final String fileName =
+        String.format(
+            "slot%s_root%s.ssz", block.getSlot(), block.getRoot().toUnprefixedHexString());
+    final Path expectedFile = tempDir.resolve("invalid_blocks").resolve(fileName);
+    checkFileNotExist(expectedFile);
+  }
+
+  @Test
+  void saveBytesToFile_shouldNotThrowExceptionWhenUnableToSave(@TempDir Path tempDir) {
```

Comment:
Could you please add a test for the IOException path as well? That is the one we should be worried about as it can be caused by unexpected reasons.